src/wp-includes/global-styles-and-settings.php

```diff
@@ -440,6 +441,36 @@ function wp_get_theme_directory_pattern_slugs() {
 	return WP_Theme_JSON_Resolver::get_theme_data( array(), array( 'with_supports' => false ) )->get_patterns();
 }
 
+/**
+ * Returns the metadata for the template parts defined by the theme.
+ *
+ * @since 6.4.0
+ *
+ * return string[]
+ */
+function wp_get_theme_template_part_metadata() {
```

Comment:
I'm not sure how I feel about this function name. The term "metadata" already has another meaning in WP, and I would prefer if we went with a name that more closely stuck to the method that it "replaces" / wraps, which is `WP_Theme_JSON_Resolver::get_theme_data()->get_template_parts()`.

Maybe `wp_get_theme_template_parts()`, or `wp_get_theme_data_template_parts()`, or `wp_theme_data_get_template_parts()`?

That name should then also be reflected in the cache key used.

Reply:
There's also another piece of template metadata to be added: `get_theme_data()->get_custom_templates()`. The reason I want to add some sort of "metadata"/"data"/etc. affix to both is that `wp_get_theme_template_parts()` can lead consumers to believe this method actually returns the template parts – which it doesn't.

What about `wp_get_theme_template_parts_data` and `wp_get_theme_custom_templates_data`?

Reply:
I went for `wp_get_theme_data_template_parts` in https://github.com/WordPress/wordpress-develop/pull/4971/commits/dc71f6e99b3a44ad083b5fa8e75fe4dfcd74ce79